In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("../data/scite-lens-merged.csv")

In [4]:
normalized = df.loc[df["supporting_cites"].ge(1) | df["disputing_cites"].ge(1)]

In [5]:
f"{normalized.shape[0]}, or {str(round((normalized.shape[0] / df.shape[0] * 100), 2))}% of pubs have equal of greater than 1 supporting/disputing cites"

'2980, or 16.14% of pubs have equal of greater than 1 supporting/disputing cites'

In [6]:
fetched_data = normalized.loc[
    normalized["first_approach"].notnull() & normalized["second_approach"].notnull() & normalized["third_approach"].notnull()
]

In [7]:
f"{fetched_data.shape[0]}, or {str(round((fetched_data.shape[0] / df.shape[0] * 100), 2))}% of pubs have equal of greater than 1 supporting/disputing cites and have been found using LENS database"

'2925, or 15.84% of pubs have equal of greater than 1 supporting/disputing cites and have been found using LENS database'

Using the same approach to definitions (and working with a subset of 2925 publications, or 15.84% of our full dataset), we now have
1. Political science or International relations or International security - 84, or 2.87% of DOIs with filled 'fields of study' column that have equal of greater than 1 supporting/disputing cites
2. International relations or International security - 8, or 0.02%
3. ((Political science or International relations or International security) AND NOT Criminal) - 70, or 0.23%

In [8]:
fetched_data["first_approach"].value_counts()

0.0    2841
1.0      84
Name: first_approach, dtype: int64

In [9]:
fetched_data["first_approach"].value_counts(normalize=True)

0.0    0.971282
1.0    0.028718
Name: first_approach, dtype: float64

In [10]:
fetched_data["second_approach"].value_counts()

0.0    2917
1.0       8
Name: second_approach, dtype: int64

In [11]:
fetched_data["second_approach"].value_counts(normalize=True)

0.0    0.997265
1.0    0.002735
Name: second_approach, dtype: float64

In [12]:
fetched_data["third_approach"].value_counts()

0.0    2855
1.0      70
Name: third_approach, dtype: int64

In [13]:
fetched_data["third_approach"].value_counts(normalize=True)

0.0    0.976068
1.0    0.023932
Name: third_approach, dtype: float64

In [14]:
fetched_data.to_csv("../data/scite-lens-15percent.csv", index=False)